persistence/src/androidMain/kotlin/com/wire/kalium/persistence/db/UserDatabaseProvider.kt

```diff
@@ -61,6 +64,8 @@ actual class UserDatabaseProvider(
                 db.execSQL("PRAGMA foreign_keys=ON;")
             }
         }
+        storePath.mkdirs()
+        cacheAssetPath.mkdirs()
```

Comment:
I decided to create them here so that we synchronise all the db / storage folders creation. These folders are used to provide assets persistent and temporary storage folders, depending on their nature & purpose.

Reply:
Couldn't they be created on demand?

Like, if I want to write a file `app/cache/file1.jpg`, the first action would be creating `app/cache/` directories if needed, and throw a failure if couldn't create them, instead of a silent failure here.

Putting this inside DB is kinda unexpected. They're just kinda hanging around.
DB essentially has nothing to do with these folders.

There's no guarantee that I will write files to this `storePath` and `cacheAssetPath`.

One could create a `UserDatabaseProvider` with a `cacheAssetPath = "app/cache"`. And then attempt to save files to `"app/files"` without any idea about what went wrong.


Also, this same operations seem to be done in `UserSessionScopeCommon.kt#452`:
```kt
    val kaliumFileSystem: KaliumFileSystem by lazy {
        // Create the cache and asset storage directories
        KaliumFileSystemImpl(dataStoragePaths).also {
            if (!it.exists(dataStoragePaths.cachePath.value.toPath()))
                it.createDirectory(dataStoragePaths.cachePath.value.toPath())
            if (!it.exists(dataStoragePaths.assetStoragePath.value.toPath()))
                it.createDirectory(dataStoragePaths.assetStoragePath.value.toPath())
        }
    }
```

Which I kinda prefer, as at least these are handled around the `FileSystem` itself, and not DB classes.